src/main/java/org/sqlite/jdbc3/JDBC3ResultSet.java

```diff
@@ -311,46 +312,21 @@ public Date getDate(int col) throws SQLException {
      * @see java.sql.ResultSet#getDate(int, java.util.Calendar)
      */
     public Date getDate(int col, Calendar cal) throws SQLException {
-        checkCalendar(cal);
-
-        switch (db.column_type(stmt.pointer, markCol(col))) {
-            case SQLITE_NULL:
-                return null;
-    
-            case SQLITE_TEXT:
-                try {
-                	FastDateFormat dateFormat = FastDateFormat.getInstance(stmt.conn.dateStringFormat, cal.getTimeZone());
-
-                    return new java.sql.Date(dateFormat.parse(db.column_text(stmt.pointer, markCol(col))).getTime());
-                }
-                catch (Exception e) {
-                    SQLException error = new SQLException("Error parsing time stamp");
-                    error.initCause(e);
-
-                    throw error;
-                }
-    
-            case SQLITE_FLOAT:
-                return new Date(julianDateToCalendar(db.column_double(stmt.pointer, markCol(col)), cal).getTimeInMillis());
-    
-            default: // SQLITE_INTEGER: 
-                cal.setTimeInMillis(db.column_long(stmt.pointer, markCol(col)) * stmt.conn.dateMultiplier);
-                return new Date(cal.getTime().getTime());
-        }
+        return getDate(col);
     }
 
     /**
      * @see java.sql.ResultSet#getDate(java.lang.String)
      */
     public Date getDate(String col) throws SQLException {
-        return getDate(findColumn(col), Calendar.getInstance());
+        return getDate(findColumn(col));
     }
 
     /**
      * @see java.sql.ResultSet#getDate(java.lang.String, java.util.Calendar)
      */
     public Date getDate(String col, Calendar cal) throws SQLException {
-        return getDate(findColumn(col), cal);
```

Comment:
(Duplicate) It seems like this moves the driver away from the JDBC spec by ignoring the user supplied Calendar/Timezone information.